adapters/nextmillennium/nextmillennium.go

```diff
@@ -0,0 +1,162 @@
+package nextmillennium
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type NextMillenniumBidRequest struct {
+	ID   string `json:"id"`
+	Test uint8  `json:"test,omitempty"`
+	Ext  struct {
+		Prebid struct {
+			StoredRequest struct {
+				ID string `json:"id"`
+			} `json:"storedrequest"`
+		} `json:"prebid"`
+	} `json:"ext"`
+}
+
+//MakeRequests prepares request information for prebid-server core
+func (adapter *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	resImps, err := getImpressionsInfo(request.Imp)
+	if len(resImps) == 0 {
+		return nil, err
+	}
+
+	result := make([]*adapters.RequestData, 0, len(resImps))
+	for _, imps := range resImps {
+		bidRequest, err := adapter.buildAdapterRequest(request, imps, request.Imp)
+		if err != nil {
+			return nil, []error{err}
+		} else {
+			result = append(result, bidRequest)
+		}
+	}
+	return result, nil
+}
+
+// getImpressionsInfo checks each impression for validity and returns impressions copy with corresponding exts
+func getImpressionsInfo(imps []openrtb2.Imp) (resImps []*openrtb_ext.ImpExtNextMillennium, errors []error) {
+	for _, imp := range imps {
+		impExt, err := getImpressionExt(&imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		if err := validateImpression(impExt); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		resImps = append(resImps, impExt)
+	}
+
+	return
+}
+
+func validateImpression(impExt *openrtb_ext.ImpExtNextMillennium) error {
+	if impExt.PlacementID == "" {
+		return &errortypes.BadInput{Message: "No valid placement provided"}
+	}
+	return nil
+}
+
+func getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ImpExtNextMillennium, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	var NextMillenniumExt openrtb_ext.ImpExtNextMillennium
```

Comment:
Nitpick: Variables should begin with a lower case letter per Go style guides.

Reply:
@SyntaxNode Done!